doc/git-subrepo.swim

```diff
@@ -599,6 +599,74 @@ If you want to chat about the `git-subrepo` command, join `#gitcommands` on
 * Written in (very modern) Bash, with full test suite. Take a look.
 * A `.gitrepo` file never is in the top level dir (next to a `.git/` dir).
 
+= Working with nested subrepos
+
+`git-subrepo` supports adding subrepos which themselves make use of
+subrepos. In fact, subrepo itself is built this way. Take a look in the
+`ext/` folder, it houses 2 dependencies as subrepos, `test-more-bash` and
+`bashplus`. 
+
+`test-more-bash` itself depends on 2 more subrepos: `bashplus` again, and
+`test-tap-bash`.
+
+The structure is therefore as follows:
+
+  git-subrepo
+    |- bashplus
+    |- test-more-bash
+      |- bashplus
+      |- test-tap-bash
+
+However, it's important to understand how this works. Here's the key idea:
+*`git-subrepo` is 100% agnostic about your dependencies!*
+
+There is *no* special handling for nested subrepos. When you clone a
+subrepo, *all* `git-subrepo` does is download the code and set up the subrepo
+file. It couldn't care less about the actual contents of your code.
+This means that all the subrepo commands simply act on the subrepo as a whole,
+treating nested subrepos like it treats any other part of the source code. 
+
+Let's look at what this means in practice. Say you have a structure as follows:
+
+  app-foo
+    |- barlib
+       |- bazlib
+
+You're working on app-foo, and make some changes to bazlib. How should you
+upstream these changes?
+
+Well, first you must realize that from the perspective of app-foo, you've simply
```

Comment:
I try to avoid using phrases such as "first you must realize". Even if it is true the fact that they are reading the documentation means that the documentation will educate them to the level they will realized these facts when they are done.

Reply:
I don't see being addressed in the commits on this PR.